src/components/JuzMarker/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { renderToString } from 'react-dom/server';
```

Comment:
This should not be needed as this can be rendered on the client

Reply:
@mmahalwy  dangerouslySetInnerHtml only takes in valid strings and not raw components
i need it to convert a certain component to a string and pass it down to dangerouslySetInnerHtml

Reply:
passing the component directly returns an error. Something about it having to be a string

Reply:
Can you paste the error? I can help more that way. But I am pretty confident we should not be using this on the client

Reply:
Asalamu Aleikum @mmahalwy 
<img width="1440" alt="screen shot 2018-03-27 at 11 41 56 am" src="https://user-images.githubusercontent.com/32780926/37956529-466fba2e-31b4-11e8-9dee-fd27d87fb6d8.png">
The above is what i get when i remove the renderToString Function
     ``` dangerouslySetInnerHTML={{
        __html: <Decoration juzNumber={juzNumber} />
      }}```
I decided to do abit more research based on what you said earlier
<cite>But I am pretty confident we should not be using this on the client</cite>
and 've got a bit of an **exciting news**
The react docs says: Link (Here)[https://reactjs.org/docs/react-dom-server.html]
`renderToString()
ReactDOMServer.renderToString(element)
Render a React element to its initial HTML. React will return an HTML string. You can use this method to generate HTML on the server and send the markup down on the initial request for faster page loads and to allow search engines to crawl your pages for SEO purposes.`

Reply:
I see why you're getting what you got in the screenshot, it's because you are turning a javascript class into a string, that's what `dangerouslySetInnerHTML` will do. Instead, `<Decoration juzNumber={juzNumber} />` should be a child of `<b />`. 

So:
```
<b
          {...bindTooltip}
          id={`Juz-mark-${chapterId}-${verseNumber}`}
          key={`Juz-${juzNumber}`}
          className="col-xs-1"
          title={`${intl.formatMessage({
            id: 'juz.index.heading',
            defaultMessage: 'juz'
          })} ${juzNumber}`}
        >
        <Decoration juzNumber={juzNumber} />
     </b>
```
should work